orderer/consensus/smartbft/chain.go

```diff
@@ -321,6 +326,37 @@ func (c *BFTChain) Deliver(proposal types.Proposal, signatures []types.Signature
 	c.support.WriteBlock(block, nil)
 }
 
+func (c *BFTChain) reportIsLeader(proposal *types.Proposal) {
+	var nodes []uint64
+	for _, n := range c.RemoteNodes {
+		nodes = append(nodes, n.ID)
+	}
+	nodes = append(nodes, c.Config.SelfID)
+	sort.Slice(nodes, func(i, j int) bool {
```

Comment:
just sort it when we construct the chain object, and then we can only do it once.